api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java

```diff
@@ -80,7 +81,7 @@ public String name() {
   public TableIdentifier toLowerCase() {
     String[] newLevels =
         Arrays.stream(namespace().levels()).map(String::toLowerCase).toArray(String[]::new);
-    String newName = name().toLowerCase();
+    String newName = name().toLowerCase(Locale.ROOT);
```

Comment:
I think we should also update `TestTableIdentifier`?

Reply:
`TestTableIdentifier` calls `org.apache.iceberg.catalog.TableIdentifier#toLowerCase`
i did not find calls to `String#toLowerCase` in `TestTableIdentifier`

Reply:
thanks for confirming, I was just quickly checking all codeplaces and didn't realize that it's using `TableIdentifier`